### ISSUE TITLE
FR: Passing command line arguments to lua interpreter

### ISSUE THREAD

**Issue by fabian-thomas** [lsyncd/lsyncd#715]
For my use case (see below), I need a way to pass custom arguments to the lua interpreter. What is your opinion on adding something like this? I already skimmed through https://github.com/lsyncd/lsyncd/issues/490, and it seemed like you are currently rather proposing to remove command line flags, in favor of doing everything in lua.
Note that I'm completely fine if this is closed as wont fix, since my workaround (see below) works like a charm.

### My use case
I usually have one configuration file that I use for all my machines. The only change I make (when I initialize the config for another machine) is the machine to deploy to. My config can be broken down to:
``` lua
...
local basename = string.match(pwd, "/([^/]+)$")
local mysync = {
    target = "machine01:" .. basename,
}
...
```

It would be easier for me to just have one configuration file and do something like:
`lsyncd myconfig.lua --arg machine machine01`, where `machine="machine01"` would be set in the lua interpreter. (much like jq does it with the `--arg` flag)

### Convenient workaround
I can rely on environment variables to achieve what I need:
```lua
local machine=os.getenv("MACHINE")
```

Then add a wrapper script (I call it start-sync):
```sh
#!/usr/bin/env sh
[ -z "$1" ] && exit 1
export MACHINE="$1"
shift 1
lsyncd sync.lua "$@"
```

Use like: `start-sync machine01`

**Comment by axkibe** [comment]
I believe with requiring luaposix you should have access to process.argv. But even better you can directly access the hostname of the machine you are running on.

**Comment by axkibe** [comment]
PS the thing you cited was my opinion a while back. But it doesn't matter anymore, since I'm no longer actively developing. poelzi is maintaining now.

**Comment by fabian-thomas** [comment]
> I believe with requiring luaposix you should have access to process.argv.

But lsyncd will complain about additional args, right? Otherwise, that might work.

>  But even better you can directly access the hostname of the machine you are running on.

Ah, my description might be confusing. I always run lsyncd on my local machine, and then decide on which machine I want to work one.

**Comment by axkibe** [comment]
Yes, it will complain, but I dont think its a complicated hack to make it ignore arguments, if you are willing to do that.
Or you create symlinks of the config file or even the lsyncd binary and use luaposix to inspect process.argv to figure out what symlink was used.

**Comment by fabian-thomas** [comment]
If @poelzi or you would merge that upstream, I would certainly try to implement it. Otherwise I would just stick to my current solution with environment variables, since that works out of the box.
